pkg/argocd/applications.go

```diff
@@ -0,0 +1,196 @@
+package argocd
+
+import (
+	"context"
+	"fmt"
+
+	argocd "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ApplicationBuilder struct {
+	Definition *argocd.Application
+	Object     *argocd.Application
+	apiClient  *clients.Settings
+	errorMsg   string
+}
+
+// Pull pulls existing ...
+func PullApplication(apiClient *clients.Settings, name, nsname string) (*ApplicationBuilder, error) {
+	glog.V(100).Infof("Pulling existing Application name %s under namespace %s from cluster", name, nsname)
+
+	builder := ApplicationBuilder{
+		apiClient: apiClient,
+		Definition: &argocd.Application{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the Application is empty")
+
+		builder.errorMsg = "Application 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the Application is empty")
+
+		builder.errorMsg = "Application 'namespace' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("application object %s doesn't exist in namespace %s", name, nsname)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Exists checks whether the given argocd application exists.
+func (builder *ApplicationBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if argocd app %s exists in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Get returns argocd application object if found.
+func (builder *ApplicationBuilder) Get() (*argocd.Application, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting argocd app %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	argocd := &argocd.Application{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name:      builder.Definition.Name,
+		Namespace: builder.Definition.Namespace,
+	}, argocd)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return argocd, err
+}
+
+// validate will check that the builder and builder definition are properly initialized before
+// accessing any member fields.
+func (builder *ApplicationBuilder) validate() (bool, error) {
```

Comment:
Please move private method to the end of file. Better to keep all public methods on top of file